src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -4245,6 +4245,91 @@ public void XA1018 ()
 			}
 		}
 
+		static readonly object [] XA1025XA1026Source = new object [] {
+			new object [] {
+				/* linkTool */                   "r8",
+				/* enableProguard */             null,
+				/* androidEnableProguard */      "true",
+			},
+			new object [] {
+				/* linkTool */                   "proguard",
+				/* enableProguard */             null,
+				/* androidEnableProguard */      "true",
+			},
+			new object [] {
+				/* linkTool */                   null,
+				/* enableProguard */             null,
+				/* androidEnableProguard */      null,
+			},
+			new object [] {
+				/* linkTool */                   null,
+				/* enableProguard */             "true",
+				/* androidEnableProguard */      null,
+			},
+			new object [] {
+				/* linkTool */                   null,
+				/* enableProguard */             null,
+				/* androidEnableProguard */      "true",
+			}
+		};
+
+		[Test]
+		[TestCaseSource (nameof (XA1025XA1026Source))]
+		public void XA1025XA1026 (string linkTool, string enableProguard, string androidEnableProguard)
+		{
+			string warning;
+			var proj = new XamarinAndroidApplicationProject {
+				// Make sure the test covers the scenario where `$(AndroidDexTool)` is not explicitly configured
+				DexTool = null,
+				LinkTool = linkTool
+			};
+			proj.SetProperty ("EnableProguard", enableProguard);
+			proj.SetProperty ("AndroidEnableProguard", androidEnableProguard);
+			using (var builder = CreateApkBuilder ()) {
+				builder.Target = "_CheckNonIdealConfigurations";
+				if (linkTool == "r8") {
```

Comment:
In other tests rather than having these kinds of `if/else` blocks, we include the expected results in the input parameters and in the `TestCaseSource` arguments.  e.g

```
new object [] {
 	/* linkTool */                   null,
 	/* enableProguard */             null,
 	/* androidEnableProguard */      "true",
        /* expectedBuildResult */ false,
        /* expectNumWarnings */ 0,
       /* expectedWarningCode */ "XA1025",                       
}
```

This would allow you to collapse all that also identical code into maybe one code block.

Also you should be able make the `androidEnableProguard` a nullable bool. 'bool? androidEnableProguard' so you can do an actual boolean evaluation rather than doing a string comparison.

Reply:
A good example is here https://github.com/xamarin/xamarin-android/blob/c477f1175b9b07fe4b6d4765aecb011dea1f0db8/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/AndroidRegExTests.cs#L170

Reply:
Very nice. Applied that change!